docs/transfer/README.md

```diff
@@ -1,42 +1,137 @@
+
+If the User has linked more than 1 account with the PISP application, the PISP application can ask the user to choose an account they wish to send funds from. Upon confirming the _source of funds_ account, the PISP can determine:
+1. the `FSPIOP-Destination` as the DFSP who the User's account is linked with
+2. The `payer` field of the `POST /thirdpartyRequests/transactions` request body. The `partyIdType` is `THIRD_PARTY_LINK`, the `fspId` is the fspId of the DFSP who issued the link, and the `partyIdentifier` is the `accountId` specified in the `POST /consents#scopes` body. 
+
+> See [1.5 Grant Consent](../linking/README.md#15-grant-consent) for more information.
+
+The PISP then generates a random `transactionRequestId` of type UUID (see [RFC 4122 UUID](https://tools.ietf.org/html/rfc4122)).
+
+![1-2-1-agreement](../out/transfer/1-2-1-agreement.svg)
+
+Upon receiving the `POST /thirdpartyRequests/transactions` call from the PISP, the DFSP performs some validation such as:
+1. Determine that the `payer` identifer exists, and is one that was issued by this DFSP to the PISP specified in the `FSPIOP-Source`.
+2. Confirms that the `Consent` that is identified by the `payer` identifier exists, and is valid.
+3. Confirm that the User's account is open and holds enough funds to complete the transaction.
+4. Any other validation that the DFSP wishes to do.
+
+Should this validation succeed, the DFSP will generate a unique `transactionId` for the request, and call `PUT /thirdpartyRequests/transactions/{ID}` with this `transactionId` and a `transactionRequestState` of recieved. 
+
+This call informs the PISP that the Thirdparty Transaction Request was accepted, and informs them of the final `transactionId` to watch for at a later date.
+
+If the above validation fail, the DFSP should send a `PUT /thirdpartyRequests/transactions/{ID}/error` call to the PISP, with an error message communicating the failure to the PISP. See [Error Codes](../error_codes.md) for more information.
+
+#### <a name='ThirdpartyAuthorizationRequest'></a>1.2.2 Thirdparty Authorization Request
+
+The DFSP will then issue a quotation request (`POST /quotes`) to the Payee DFSP. Upon receiving the `PUT /quotes/{Id}` callback from the Payee DFSP, the Payer DFSP needs to confirm the details of the transaction with the PISP.
 
-2. **Agreement**
-    In this phase, the PISP confirms the Payee Party, and looks up the terms of the transaction. If the User accepts the terms of the transaction, they sign the transaction with the credential established in the Linking API flow
+They use the API call `POST /thirdpartyRequests/authorizations`. The request body is populated with the following fields:
 
-3. **Transfer**
-    In this phase, the Payer DFSP initiates the transaction, and informs the PISP of the transaction result.
+- `authorizationRequestId` - a random UUID generated by the DFSP to identify this Thirdparty Authorization Request 
```

Comment:
what about `transactionRequestId`, so PISP can select a proper device to sign the challenge?

Reply:
Good point